charters/ingestion-service/charter.md

```diff
@@ -0,0 +1,166 @@
+
+### Dependencies
+
+The Ingestion Service will be as loosely coupled as possible to schemas defined by the metadata team. The definition of a core domain model for data and metadata is therefore in scope for the Ingestion Service, as this core domain model must be shared between the Ingestion Service and Metadata Schema. This core domain model defines high level concepts (e.g. 'Biomaterial', 'Protocol', 'File'), and changes to this core model must be agreed between the responsible teams.
+
+All services and tools that are used by direct data contributors will be rigorously tested with users to ensure a high quality user experience, and will take advantage of user experience experts within the DCP team.
+
+### Scientific "guardrails"
+
+The Ingestion Service requires oversight on which data types are required within the DCP (for example, SmartSeq2 and 10x sequencing data, imaging data) and by extension which file formats are endorsed and acceptable within the DCP (e.g. FASTQ, BAM, TIFF).
+
+## Out-of-scope
+
+### Ingest API
+ * Metadata specification definitions. Ingest will utilise metadata definitions defined by DCP-compliant JSON schemas, as defined by the metadata working group, and will ensure the Ingest API is schema-agnostic where possible, but will not actively define metadata specifications beyond the restrictions placed by the shared core domain model outlined above
+ * Definition of bundle structures for the DCP datastore. Whilst the Ingest API will provide support for generating bundles in response to submissions, the Ingestion Service is not responsible for the specification of bundle structures - it is expected these will be provided either by the metadata team or to address specific consumer requirements
+ * Authentication services. The Ingestion Service will utilise DCP-wide authentication services. Authorisation services for specific ingest requirements are likely out of scope, unless DCP-wide authorisation services cannot fulfil requirements that are overly specific to ingest
+
+### Ingest UI
+ * Preview illustration of data, for example as it "will appear" in the data portal
+ * Searching over submitted data, beyond that required for retrieval of a given user's previous submissions
+ * Dedicated views over the data, for example visualisation of all data for a particular project or from a particular tissue type
+
+### Spreadsheet Template Generator
+ * Generation of lab specific spreadsheets that are not directly generated from a JSON schema endorsed by the metadata team. Additional customisations (for example, modifying column names or adding additional tabs) for individual contributors will not be supported through any mechanism other than through the use of standard configuration (i.e. which modules are to be included) or through the proposal of metadata schema changes
```

Comment:
One comment I have is how does Ingest deal with data that is either 1) not well supported and bleeding edge and 2) has metadata specifications but is currently not being processed by Green.  Seems like 2 should definitely but uploadable but not sure about 1.  Is there a path to submitting data (with, say, minimal metadata) to get it in blue and available in orange even before the full metadata model is not yet prepared?  This overlaps with the metadata charter too but I'd like to hear about it from the perspective of ingest.

Reply:
Yes, great question. The ingest perspective is that anything with a defined metadata schema that is a valid JSON schema can be uploaded and will be validated, and if valid will be deposited in the DSS. I think this covers your scenario 2.

It is also possible, if you have truly bleeding edge data (scenario 1), to add additional metadata fields (e.g. creating new columns in the spreadsheet) to push data to ingest. This won't validate against the current schema (and therefore won't be released to the DSS), but we have provision to revalidate once a new schema is published. Of course, the alternative is that you create a submission that validates against the last schema but e.g. misses some new fields needed, and then providing an update once the new schema is released. Updates need to be supported before we can deliver on this but this is the plan.

Hopefully I've explained that clearly? Most of this functionality is already supported in the spreadsheet template importer, and for programmatic submissions we *always* retain whatever is sent even if it's invalid

Reply:
Do you have a suggestion for how I can update the charter to help communicate this better?